eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentPacket.scala

```diff
@@ -264,7 +264,10 @@ object OutgoingPaymentPacket {
 
   /** Build the command to add an HTLC for the given recipient using the provided route. */
   def buildOutgoingPayment(replyTo: ActorRef, upstream: Upstream, paymentHash: ByteVector32, route: Route, recipient: Recipient): Try[OutgoingPaymentPacket] = {
-    val outgoingChannel = route.hops.head.shortChannelId
+    val outgoingChannel = route.hops.head match {
+      case hop: ChannelHop => hop.shortChannelId
+      case _: BlindedHop => ???
```

Comment:
Implementation missing.

Reply:
That's on purpose, it's not necessary to make that version of the code work since this is dead code (the code path cannot be triggered). It's done this way to split the implementation work between this PR and the next one, where this is actually implemented.